src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -75,11 +75,14 @@ public Double quotaLimit(ClientQuotaType quotaType, Map<String, String> metricTa
      * Put a small delay between logging
      */
     private void maybeLog(AtomicLong latLoggedMessageTimeMs, String format, Object... args) {
-        long now = System.currentTimeMillis();
-        if (now - latLoggedMessageTimeMs.get() >= LOGGING_DELAY_MS) {
-            log.debug(format, args);
-            latLoggedMessageTimeMs.set(now);
-        }
+        latLoggedMessageTimeMs.getAndUpdate(current -> {
+            long now = System.currentTimeMillis();
+            if (now - current >= LOGGING_DELAY_MS) {
+                log.debug(format, args);
```

Comment:
The lambda can get called multiple times (read up about compare and swap to understand how), so use a `final boolean[]` flag to pass whether to log from the lambda to the code following `getAndUpdate`.